cloud/ec2.go

```diff
@@ -1406,13 +1398,9 @@ func (m *ec2Manager) AssociateIP(ctx context.Context, h *host.Host) error {
 	return errors.Wrapf(associateIPAddressForHost(ctx, m.client, h), "associating allocated IP address '%s' with host '%s'", h.IPAllocationID, h.Id)
 }
 
-// CleanupIP disassociates the IP address from the host's network interface and
-// releases the IP address back into the IPAM pool.
+// CleanupIP releases the host's IP address.
 func (m *ec2Manager) CleanupIP(ctx context.Context, h *host.Host) error {
-	if err := disassociateIPAddressForHost(ctx, m.client, h); err != nil {
-		return err
-	}
-	if err := releaseIPAddressForHost(ctx, m.client, h); err != nil {
+	if err := releaseIPAddressForHost(ctx, h); err != nil {
```

Comment:
nit: can just `return releaseIPAddressForHost(ctx, h)`

Reply:
So true, updated this + the fleet implementation.